src/boot/efi/smbios.c

```diff
@@ -181,3 +201,44 @@ const char* smbios_find_oem_string(const char *name) {
 
         return NULL;
 }
+
+static const char* smbios_get_string(const SmbiosHeader *header, size_t nr, size_t left) {
+        assert(header);
+        const char *s = (const char *) header;
+
+        assert(left >= header->length);
+        s += header->length;
+        left -= header->length;
+
+        size_t index = 1;
+        for (const char *p = s; index <= nr && p < s + left; index++) {
+                const char *e = memchr(p, 0, s + left - p);
+                if (!e || e == p) /* Double NUL byte means we've reached the end of the strings. */
+                        break;
+
+                if (index == nr)
+                        return p;
+
+                p = e + 1;
+        }
+        return NULL;
+}
+
+void smbios_raw_info_populate(RawSmbiosInfo *ret_info) {
+        assert(ret_info);
+        size_t left;
+
+        const SmbiosTableType1 *type1 = (const SmbiosTableType1 *) get_smbios_table(1, &left);
+        if (type1 && type1->header.length >= sizeof(SmbiosTableType1) && left >= type1->header.length) {
```

Comment:
please rebase n top of #34718 so that all you need to do is check `type1` being set, the other two hecks can then go